src/lib/shell/commands/Device.cpp

```diff
@@ -42,7 +43,7 @@ int DeviceHelpHandler(int argc, char ** argv)
 static CHIP_ERROR FactoryResetHandler(int argc, char ** argv)
 {
     streamer_printf(streamer_get(), "Performing factory reset ... \r\n");
-    DeviceLayer::ConfigurationMgr().InitiateFactoryReset();
+    chip::Server::GetInstance().ScheduleFactoryReset();
```

Comment:
This was done this way to avoid dependency between `src/lib` and `app` as `app` was supposed to be the top-level component. I guess the best way to workaround it would be moving the shell commands to the `app` directory but not sure how much work that would be...

Reply:
Ah, okay. Closing the PR